test_analysis3.py

```
@@ -0,0 +1,41 @@
+    rel_path = os.path.relpath(lib_file, lib_dir)
+    test_file_path = os.path.join(test_dir, rel_path.replace(".dart", "_test.dart"))
+
+    if test_file_path not in test_files:
+        untested_files.append(lib_file)
+
+# Look for high value untested files (blocs, repositories, usecases)
+high_value_untested = []
+for f in untested_files:
+    # only include abstract files
+    if ("bloc" in f or "cubit" in f or "repository" in f or "usecase" in f or "service" in f or "controller" in f or "provider" in f or "notifier" in f or "model" in f or "entity" in f):
+        if not f.endswith("_state.dart") and not f.endswith("_event.dart") and "repository.dart" in f:
```

Ruff (0.15.7)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)